gazebo_ros/src/node.cpp

```diff
@@ -164,4 +183,22 @@ rclcpp::Logger Node::internal_logger()
   return rclcpp::get_logger("gazebo_ros_node");
 }
 
+void NodeLookUp::add_node(const std::string & node_name)
+{
+  std::lock_guard<std::mutex> guard(this->internal_mutex_);
+  this->set_.insert(node_name);
+}
+
+void NodeLookUp::remove_node(const std::string & node_name)
+{
+  std::lock_guard<std::mutex> guard(this->internal_mutex_);
+  this->set_.erase(node_name);
+}
+
+bool NodeLookUp::is_node_name_in_set(const std::string & node_name)
+{
+  std::lock_guard<std::mutex> guard(this->internal_mutex_);
+  return this->set_.find(node_name) != this->set_.end();
+}
```

Comment:
I'd prefer the class name to be more descriptive, say ``ExistingNodes``, and the ``is_node_name_in_set`` could be called ``check_node``.